### PR TITLE
Fix/copyfrom context

### PR DESCRIPTION
**Pull request by dzhwinter**
fix https://github.com/PaddlePaddle/Paddle/issues/6948

### CODE REVIEW

File: paddle/framework/tensor_util.h [hunk 1]
```diff
@@ -82,6 +82,28 @@ inline void CopyFrom(const Tensor& src, const platform::Place& dst_place,
 #endif
 }
 
+/**
+ * @brief CopyFrom support CPU <-> CPU
+ */
+
```
Comment (Yancey1989) [inline paddle/framework/tensor_util.h]: Delete the redundant black line.

File: paddle/framework/tensor_util.h [hunk 2]
```diff
@@ -82,6 +82,28 @@ inline void CopyFrom(const Tensor& src, const platform::Place& dst_place,
...
+  dst->Resize(src.dims());
+
+  auto src_place = src.place();
+  auto src_ptr = src.data<void>();
+
+  auto dst_ptr = dst->mutable_data(dst_place, src.type());
+
+  auto size = src.numel() * SizeOfType(src.type());
+
+  if (platform::is_cpu_place(src_place) && platform::is_cpu_place(dst_place)) {
```
Comment (Yancey1989) [inline paddle/framework/tensor_util.h]: How about use `PADDLE_ENFORCE` to checkt the place?

File: paddle/framework/tensor_util.h [hunk 3]
```diff
@@ -115,6 +137,21 @@ inline void CopyFromVector(const std::vector<T>& src,
...
+ */
+template <typename T>
+inline void CopyFromVector(const std::vector<T>& src, Tensor* dst) {
+  platform::CPUPlace dst_place = platform::CPUPlace();
+  auto src_ptr = static_cast<const void*>(src.data());
+  platform::CPUPlace src_place;
+  dst->Resize({static_cast<int64_t>(src.size())});
+  auto dst_ptr = static_cast<void*>(dst->mutable_data<T>(dst_place));
+  auto size = src.size() * sizeof(T);
+
```
Comment (Yancey1989) [inline paddle/framework/tensor_util.h]: Add place checker.

File: paddle/framework/tensor_util.h (hunk 2 above)
Comment (dzhwinter) [reply paddle/framework/tensor_util.h]: Done.

File: paddle/framework/tensor_util.h (hunk 1 above)
Comment (dzhwinter) [reply paddle/framework/tensor_util.h]: Done.

File: paddle/framework/tensor_util.h (hunk 3 above)
Comment (dzhwinter) [reply paddle/framework/tensor_util.h]: CopyToVector create a cpu_place inside. No need to check it.

**Pull request merged**